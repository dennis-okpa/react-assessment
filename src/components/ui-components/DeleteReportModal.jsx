/* eslint-disable react/no-array-index-key */
import React from 'react';
import {
  YesBtn,
  NoBtn,
  Select,
  Plead
} from '@components/styled-components/Task';
import Modal from './Modal';
import { options } from '../../assets/constants';

const DeleteReportModal = () => (
  <Modal
    title="Are you sure you want to delete this report and its history?"
    explanation={(
      <>
        If you delete the
        {' '}
        <strong>Executive metrics</strong>
        {' '}
        report, you will also delete the associated history:
      </>
    )}
    actionPanel={(
      <>
        <YesBtn onClick={(e) => { e.stopPropagation(); console.log('deleting all'); }}>Delete all</YesBtn>
        <NoBtn onClick={(e) => { e.stopPropagation(); console.log('canceling job'); }}>Cancel</NoBtn>
      </>
    )}
  >
    <Select multiple>
      {options.map((option, i) => <option key={i}>{option}</option>)}
    </Select>
    <Plead>
      Please type the word &lsquo;Delete&rsquo; to remove the
      {' '}
      <strong>Executive metrics</strong>
      {' '}
      report and its associated history:
    </Plead>
  </Modal>
);

export default DeleteReportModal;
