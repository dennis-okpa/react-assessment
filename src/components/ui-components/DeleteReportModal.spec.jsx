import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import DeleteReportModal from './DeleteReportModal';

const mockLog = jest.spyOn(global.console, 'log').mockImplementation();

describe('The DeleteReportModal Component', () => {
  describe('when rendered', () => {
    let container;
    let getByText;

    beforeEach(() => {
      ({ container, getByText } = render(
        <DeleteReportModal />
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should call console with the correct string when Delete all is clicked', async () => {
      await act(async () => {
        fireEvent.click(getByText('Delete all'));
      });

      expect(mockLog).toHaveBeenCalledWith('deleting all');
    });

    it('should call console with the correct string when Cancel is clicked', async () => {
      await act(async () => {
        fireEvent.click(getByText('Cancel'));
      });

      expect(mockLog).toHaveBeenCalledWith('canceling job');
    });
  });
});
