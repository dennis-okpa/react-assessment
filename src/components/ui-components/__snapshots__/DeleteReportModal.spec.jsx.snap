// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DeleteReportModal Component when rendered should match snapshot 1`] = `
<div>
  <div
    class="Task__Panel-sc-7z955n-6 diyQSp"
  >
    <div>
      <h3>
        Are you sure you want to delete this report and its history?
      </h3>
      <hr />
      <span>
        If you delete the
         
        <strong>
          Executive metrics
        </strong>
         
        report, you will also delete the associated history:
      </span>
    </div>
    <select
      class="Task__Select-sc-7z955n-13 dPObDm"
      multiple=""
    >
      <option>
        January 2020
      </option>
      <option>
        Febuary 2020
      </option>
      <option>
        March 2020
      </option>
      <option>
        April 2020
      </option>
      <option>
        May 2020
      </option>
      <option>
        June 2020
      </option>
      <option>
        July 2020
      </option>
      <option>
        August 2020
      </option>
      <option>
        September 2020
      </option>
      <option>
        October 2020
      </option>
      <option>
        November 2020
      </option>
      <option>
        December 2020
      </option>
    </select>
    <span
      class="Task__Plead-sc-7z955n-14 erYGTz"
    >
      Please type the word ‘Delete’ to remove the
       
      <strong>
        Executive metrics
      </strong>
       
      report and its associated history:
    </span>
    <div
      class="Task__ActionPanel-sc-7z955n-9 jDSgIL"
    >
      <button
        class="Task__ActionBtn-sc-7z955n-10 Task__YesBtn-sc-7z955n-11 iozTJj bgrwqn"
      >
        Delete all
      </button>
      <button
        class="Task__ActionBtn-sc-7z955n-10 Task__NoBtn-sc-7z955n-12 iozTJj dxGCKi"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
